### ISSUE TITLE
Can't add headers to an endpoint

### ISSUE THREAD

**Issue by kevinrenskers** [gonzalezreal/SimpleNetworking#6]
I can't do something like this:

```
extension Endpoint where Output == Void {
  static func submitChallenge(challenge: Challenge, response: String) -> Endpoint {
    let payload = ChallengePayload(code: response)
    return Endpoint(method: .patch, path: "auth/challenges/\(challenge.id)", headers: [.accept: ContentType.json.rawValue, .contentType: ContentType.json.rawValue], body: payload)
  }
}
```

Even though the server doesn't give a response for this particular endpoint, I still need it to have the `Accept: application/json` header, which isn't there when `Output` is `Void`.

I also can't construct the Endpoint and then set `endpoint.headers = [.accept: ContentType.json.rawValue, .contentType: ContentType.json.rawValue]`, since `headers` is a let.

**Comment by kevinrenskers** [comment]
Ah, I see that you made them public (https://github.com/gonzalezreal/SimpleNetworking/blob/master/Sources/SimpleNetworking/Endpoint.swift). But sadly you haven't released a new version of this library yet.

**Comment by gonzalezreal** [comment]
Hi @kevinrenskers,

I just published [SimpleNetworking 1.2.0](https://github.com/gonzalezreal/SimpleNetworking/releases/tag/1.2.0) to address this issue.

Thanks for the feedback!
